deepspeed/profiling/flops_profiler/profiler.py

```diff
@@ -157,10 +158,26 @@ def reset_profile(self):
         Adds or resets the extra attributes.
         """
 
+        def moe_expert_parallelism(param):
+            """
+            Return whether the layer is an MoE layer, and its expert parallelism size if so
+            """
+            prefix = 'ep_size_'
+            if getattr(param, "group_name", "").startswith(prefix):
+                return True, int(param.group_name[len(prefix):])
+            return False, 0
+
         def add_or_reset_attrs(module):
+            parameters = []
+            for p in module.parameters():
+                is_moe, expert_parallelism = moe_expert_parallelism(p)
+                parameters.append((p.numel(), is_moe, expert_parallelism))
```

Comment:
nit: `parameters = [(p.numel(), *moe_expert_parallelism(p)) for p in module.parameters()]

Or can we make `moe_expert_parallelism()` return all three?

Reply:
I had your version first but I don't think it's very readable. I'll switch to making the function return all three parameters. I don't think there's anything wrong with the current though